lib/open_telemetry_exporter/logger.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "logger"
+
+module OpenTelemetryExporter
+  class Logger
+    class << self
+      def increment(name, tags: {}, by: 1)
+        if OpenTelemetryExporter.configuration.opentelemetry_enabled?
+          begin
+            meter = OpenTelemetryExporter.configuration.opentelemetry_meter
+            raise "OpenTelemetry meter not configured" unless meter
+
+            meter.create_counter(name).add(by, attributes: convert_tags(tags))
+            by
+          rescue => e
+            log_error("OpenTelemetry error for metric '#{name}': #{e.message}")
+            log_error("Falling back to configured fallback")
+            call_fallback_increment(name, tags: tags, by: by)
+          end
+        else
+          call_fallback_increment(name, tags: tags, by: by)
+        end
```

Comment:
And perhaps better as guard clause?
```suggestion
        call_fallback_increment(name, tags: tags, by: by) unless OpenTelemetryExporter.enabled?

          meter = OpenTelemetryExporter.configuration.opentelemetry_meter
          raise "OpenTelemetry meter not configured" unless meter

          meter.create_counter(name).add(by, attributes: convert_tags(tags))
          by
        rescue => e
          log_error("OpenTelemetry error for metric '#{name}': #{e.message}")
          log_error("Falling back to configured fallback")
          call_fallback_increment(name, tags: tags, by: by)
```